lib/puppet/node/environment.rb

```diff
@@ -271,10 +271,11 @@ def modules
       seen_modules = {}
       modulepath.each do |path|
         Dir.entries(path).each do |name|
-          next if name == "." || name == ".."
+          next unless Puppet::Module.is_module?(name, path)
```

Comment:
Modules in a namespaced(puppetlabs-apache vs apache) path won't load properly. Is this what we want here?